### ISSUE TITLE
how to test  hypermorph?

### ISSUE THREAD

**Issue by daxuemi** [voxelmorph/voxelmorph#293]
**Task** (what are you trying to do/register?)

professor,sorry to disturb you again. I have tried to run [train_hypermorph.py](https://github.com/voxelmorph/voxelmorph/blob/dev/scripts/tf/train_hypermorph.py) these days. when I test the result,there is only the code of  [test.py](https://github.com/voxelmorph/voxelmorph/blob/dev/scripts/tf/test.py), I don't know whether it is to test hypermorph. it seems that it is not used for test hypermorph in my opinion beacuse HyperVxmDense is not called in test.py.
I would appreciate it if you could provide the test code for hypermorph.
  I have read the [test.py](https://github.com/voxelmorph/voxelmorph/blob/dev/scripts/tf/test.py), I don't  understand how the data is emitted in pair.txt . My test data and fixed image have been processed into. Npz format. I don't know if I need to change it. there are very few materials related to HyperMorph since it is released not so long ago. As a beginner, I am very eager to learn your latest research on HyperMorph. I will be very grateful if you can add some materials about how to test HyperMorph.
 Thank you again! I have learned a lot from your excellent work. Your rigorous attitude towards scientific research and the spirit of selfless dedication have always inspired me. thank you!

**Comment by daxuemi** [comment]
my data set are ABIDE and ADNI

**Comment by adalca** [comment]
@daxuemi, thanks for the follow up. We're working with @ahoopes to release a tutorial on hypermorph to help with this. 

For now, what would you like to do exactly? Test HyperMorph under different hyperparameters?

indeed, test.py is for standard VoxelMorph scripts, not full HyperMorph.

**Comment by daxuemi** [comment]
> @daxuemi, thanks for the follow up. We're working with @ahoopes to release a tutorial on hypermorph to help with this.
> 
> For now, what would you like to do exactly? Test HyperMorph under different hyperparameters?
> 
> indeed, test.py is for standard VoxelMorph scripts, not full HyperMorph.

thank you for your reply and sorry for not being able to describe my problem clearly.

About two years ago, I processed some data in the ABIDE dataset according to the method introduced by voxelmorph. The data was processed into **.npz** format and the size was cropped to 160*192*224. After training and testing, I got the image The dice score for registration is about 0.73, which is excellent.

Now I have trained the model with **hypermorph**. I hope to calculate the **dice score** on my test set to better judge the result of image registration. But it is a bit difficult for me to reproduce the code of dice that specifically calculates the hypermorph method, so I hope you can release the code of test_hypermorph.py (**aims to calculate dice score**) if possible.

thank you again!

**Comment by ahoopes** [comment]
Also, I just want to note that for a hypermorph model registering large 3D volumes, you will need substantially more epochs than the default of 1500. I will also add a more robust LR decay configuration to that script soon.

**Comment by daxuemi** [comment]
> 



> Also, I just want to note that for a hypermorph model registering large 3D volumes, you will need substantially more epochs than the default of 1500. I will also add a more robust LR decay configuration to that script soon.



> Yes, the code to sweep across lambda values or even optimize lambda for a specific set of images is not yet in the repository - that will be added very soon. In the meantime, you could customize the `test.py` script a bit for HyperMorph by adding a couple lines to loop over and evaluate lambda values. Something like:
> 
> ```python
> for hyp_n, hyp_lambda in enumerate(np.linspace(0, 1, 20)):
>     ...
>     hyp = np.array([[hyp_lambda]], dtype='float32')  # reformat hyperparam
>     moved = model.predict([moving, fixed, hyp])[0].squeeze()
>     ...
> ```
> 
> But yes, I should definitely add a `test_hypermorph.py` script for this. Sorry for the inconvience, but glad you're using HyperMorph!

Thank you very much for your reply!  I'm looking forward to your additional code for hypermorph and I will also add some code under your guidance. Thank you again for your excellent work on hypermorph.  All the best wishes for you professor!
